### ISSUE TITLE
Log profiler should provide field metadata events

### ISSUE THREAD

**Issue by alexrp** [mono/mono#6177]
In particular, it would be nice to have field types, names, and offsets. With this information, a complete object graph with detailed field information could be constructed from a heapshot. Additionally, once #6144 is merged, static field offsets would make it possible to narrow down exactly which field a root comes from.

**Comment by alexrp** [comment]
cc @DavidKarlas @rodrmoya

**Comment by marek-safar** [comment]
This would fit well with dev16
